src/Load_and_save.jl

```diff
@@ -149,7 +186,37 @@ function string_intent(intent)
   end
 end
 
+"""
+helper function for nifti
+creates a data dictionary for header data can be used to create a new NIfTI.NIfTI1Header when saving to a file
+"""
+function formulate_header_data_dict(nifti_image_header::NIfTI.NIfTI1Header)::Dictionaries.Dictionary
+
+  #below in the auxfile[1] and description[1] are the actual values for the field, followed by the string representation of them
+  #for qform_code and sform_code , their string representations of separate feilds such as qform_code_name and sform_code_name
+
+  header_data_keys = ["nifti_type", "dim_info", "dim", "nifti_intent", "nifti_intent_code", "datatype", "bitpix", "slice_start", "pixdim", "vox_offset", "scl_slope", "scl_inter", "slice_end", "slice_code",
+    "xyzt_units", "cal_max", "cal_min", "slice_duration", "toffset", "descrip", "aux_file", "qform_code", "qform_code_name", "sform_code", "sform_code_name", "quatern_b", "quatern_c", "quatern_d",
+    "qoffset_x", "qoffset_y", "qoffset_z", "srow_x", "srow_y", "srow_z", "intent_name"]
+  header_data_values = ["one", nifti_image_header.dim_info, nifti_image_header.dim, (nifti_image_header.intent_p1, nifti_image_header.intent_p2, nifti_image_header.intent_p3), nifti_image_header.intent_code,
+    nifti_image_header.datatype, nifti_image_header.bitpix, nifti_image_header.slice_start, nifti_image_header.pixdim, nifti_image_header.vox_offset, nifti_image_header.scl_slope,
+    nifti_image_header.scl_inter, nifti_image_header.slice_end, nifti_image_header.slice_code, nifti_image_header.xyzt_units, nifti_image_header.cal_max, nifti_image_header.cal_min,
+    nifti_image_header.slice_duration, nifti_image_header.toffset, (nifti_image_header.descrip, formulate_string(nifti_image_header.descrip)), (nifti_image_header.aux_file, formulate_string(nifti_image_header.aux_file)),
+    nifti_image_header.qform_code, formulate_xform_string(nifti_image_header.qform_code), nifti_image_header.sform_code, formulate_xform_string(nifti_image_header.sform_code),
+    nifti_image_header.quatern_b, nifti_image_header.quatern_c, nifti_image_header.quatern_d, nifti_image_header.qoffset_x, nifti_image_header.qoffset_y, nifti_image_header.qoffset_z,
+    nifti_image_header.srow_x, nifti_image_header.srow_y, nifti_image_header.srow_z, nifti_image_header.intent_name]
+
+  header_data_dict = Dictionaries.Dictionary(header_data_keys, header_data_values)
+  return header_data_dict
+end
+
+"""
+helper function for nifti
+creates a nifti_image struct which basically encapsulates all the necessary data, contains voxel data 
+"""
+function formulate_nifti_image_struct(nifti_image::NIfTI.NIVolume)::Nifti_image
 
```

Comment:
This function is empty at the moment, i will push functionality for this in the next pull request.

Reply:
ok :)